### DISCUSSION TITLE
Usage of `vm_trace!`

### DISCUSSION THREAD

**Discussion by jopemachine** [RustPython/RustPython#4162] [Q&A] [answered]
I found several `vm_trace!` macros under `vm/src` directory while debugging codes, but I don't have an idea how to use them.

I guess those are for debugging, but they didn't print anything on the console, or write file log, when I added `vm-tracing-logging` to the `features` of `vm/Cargo.toml`.

And there were some compile errors which had to fix it to compile, looks like it would be good to fix this.

**Comment by youknowone** [comment]
@fanninpm @coolreader18 do you have any idea?

**Comment by jopemachine** [comment, marked as answer]
I found the `trace!` macro in the `vm_trace!` belongs [log](https://github.com/rust-lang/log) crate.
I can find its usage in [this document](https://docs.rs/env_logger/0.9.1/env_logger/).

And I made a [PR](https://github.com/RustPython/RustPython/pull/4195) to fix the compile error.
